### PR TITLE
2.6.2 release candidate

### PR DESCRIPTION
**Pull request by bmeike**
Add test for empty dict
Fix test for LiveQuery delay

Change-Id: If308ff977e387dd475af0130249c546178133e70

### CODE REVIEW

File: lib/src/androidTest/java/com/couchbase/lite/ResultTest.java [hunk 1]
```diff
@@ -1082,4 +1082,23 @@ public void check(int n, Result r) throws Exception {
...
+        MutableDocument mDoc = new MutableDocument(doc1);
+        mDoc.setDictionary(key1, new MutableDictionary());
+        save(mDoc);
+
+        final Query query = QueryBuilder.select(SelectResult.property(key1))
+            .from(DataSource.database(db))
+            .where(Meta.id.equalTo(Expression.string(doc1)));
+
+        ResultSet results = query.execute();
+        for (Result result : results.allResults()) { result.toMap(); }
```
Comment (pasin) [inline lib/src/androidTest/java/com/couchbase/lite/ResultTest.java]: Should be assert if result.toMap() is empty?

Comment (bmeike) [reply lib/src/androidTest/java/com/couchbase/lite/ResultTest.java]: Wooo... good idea.  It is the `allResults()` that used to fail.  Still, that's a good test.  Will add.

File: lib/src/androidTest/java/com/couchbase/lite/ResultTest.java [hunk 2]
```diff
@@ -1082,4 +1082,29 @@ public void check(int n, Result r) throws Exception {
...
+        mDoc.setDictionary(key1, new MutableDictionary());
+        save(mDoc);
+
+        final Query query = QueryBuilder.select(SelectResult.property(key1))
+            .from(DataSource.database(db))
+            .where(Meta.id.equalTo(Expression.string(doc1)));
+
+        ResultSet results = query.execute();
+        for (Result result : results.allResults()) {
+            assertNotNull(results);
```
Comment (pasin) [inline lib/src/androidTest/java/com/couchbase/lite/ResultTest.java]: I think this one should be outside for-loop.

Comment (bmeike) [reply lib/src/androidTest/java/com/couchbase/lite/ResultTest.java]: duh.  Of course.  ... and the check for `result` inside.
Thanks.

File: lib/src/androidTest/java/com/couchbase/lite/LiveQueryTest.java [hunk 3]
```diff
@@ -84,17 +84,22 @@ public void testLiveQueryDelay() throws CouchbaseLiteException, InterruptedExcep
...
         createDocNumbered(13);
         createDocNumbered(14);
+        createDocNumbered(15);
+        createDocNumbered(16);
 
-        // Ya, I know...
-        Thread.sleep(1000);
-
-        assertEquals(1, value.get());
+        latch1.await(5, TimeUnit.SECONDS);
```
Comment (Sandychuang8) [inline lib/src/androidTest/java/com/couchbase/lite/LiveQueryTest.java]: Do we need to assertTrue here?
